### PR TITLE
Bug fix/smartgraph view no duplicate data

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

When creating a View-Link on SmartEdge collections, we create links on the internal underlying collections to make them readable. However we should not read directly out of the `_to_` part to keep data consistent, as done in AQL.
Reading from `_to_` could yield to document being returned twice. With this PR we remove the `_to_` link.

- [X] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [X] Backport for 3.7: https://github.com/arangodb/arangodb/pull/13027 / https://github.com/arangodb/enterprise/pull/587
- [X] Backport for 3.6: https://github.com/arangodb/arangodb/pull/13028 / https://github.com/arangodb/enterprise/pull/588

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [x] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [x] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [x] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by mchacki** [comment]
Jenkins:
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12649/

File: arangod/IResearch/IResearchViewCoordinator.cpp [hunk 1]
```diff
@@ -273,6 +273,15 @@ Result IResearchViewCoordinator::appendVelocyPackImpl(
 }
 
 Result IResearchViewCoordinator::link(IResearchLink const& link) {
+#ifdef USE_ENTERPRISE
+  {
+    auto const& name = link.collection().name();
```
Comment (gnusi) [inline arangod/IResearch/IResearchViewCoordinator.cpp:278]: If you don't mind I prefer to encapsulate filtering logic in a separate function, similar way you did for `createLink` https://github.com/arangodb/arangodb/blob/devel/arangod/IResearch/IResearchLinkHelper.cpp#L206

**Review by gnusi — APPROVED** [review]

File: arangod/IResearch/IResearchViewCoordinator.cpp (hunk 1 above)
Comment (mchacki) [reply arangod/IResearch/IResearchViewCoordinator.cpp:278]: Will do

**Comment by mchacki** [comment]
Jenkins after review is applied:
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12659/

**Pull request merged**